tests/test_test_tools/test_test_cases.py

```diff
@@ -239,6 +239,7 @@ def test_terminate_subprocesses(self):
         proc = self.start_subprocess("-c", "import time; time.sleep(10)")
         assert proc.returncode is None
         self._terminate_subprocesses()
+        self.subprocesses.clear()
```

Comment:
this fixes the various flaky failures in `ERROR tests/test_test_tools/test_test_cases.py::TestGenericCases` observed on all platforms,